src/__swaps__/screens/Swap/hooks/useSearchCurrencyLists.ts

```diff
@@ -23,156 +22,161 @@ export interface AssetToBuySection {
   id: AssetToBuySectionId;
 }
 
-const filterBridgeAsset = ({ asset, filter = '' }: { asset?: SearchAsset; filter?: string }) =>
+const filterBridgeAsset = ({ asset, filter = '' }: { asset: SearchAsset | null | undefined; filter: string }) =>
   asset?.address?.toLowerCase()?.startsWith(filter?.toLowerCase()) ||
   asset?.name?.toLowerCase()?.startsWith(filter?.toLowerCase()) ||
   asset?.symbol?.toLowerCase()?.startsWith(filter?.toLowerCase());
 
 export function useSearchCurrencyLists() {
   const { internalSelectedInputAsset: assetToSell, selectedOutputChainId } = useSwapContext();
 
-  const searchQuery = userAssetsStore(state => state.searchQuery);
+  const query = useSwapsStore(state => state.outputSearchQuery.trim().toLowerCase());
 
-  const [inputChainId, setInputChainId] = useState(assetToSell.value?.chainId ?? ChainId.mainnet);
-  const [toChainId, setToChainId] = useState(selectedOutputChainId.value);
-  const [assetToSellAddress, setAssetToSellAddress] = useState(
-    assetToSell.value?.[assetToSell.value?.chainId === ChainId.mainnet ? 'address' : 'mainnetAddress']
-  );
+  const [state, setState] = useState({
+    assetToSellAddress: assetToSell.value?.[assetToSell.value?.chainId === ChainId.mainnet ? 'address' : 'mainnetAddress'],
+    fromChainId: assetToSell.value?.chainId ?? undefined,
+    isCrosschainSearch: assetToSell.value ? assetToSell.value.chainId !== selectedOutputChainId.value : false,
+    toChainId: selectedOutputChainId.value ?? ChainId.mainnet,
+  });
```

Comment:
what was the reason these were combined into a single state object? Generally a bigger fan of keeping stuff separated. See https://legacy.reactjs.org/docs/hooks-faq.html#should-i-use-one-or-many-state-variables